modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,237 @@ public boolean updateResource(String applicationKey, String resourceType, String
+            for (ServiceProxy serviceProxy : serviceProxyList) {
+                addProxiesPreparedStatement.setString(1, serviceProxy.getServiceName());
+                addProxiesPreparedStatement.setString(2, serviceProxy.getServiceProtocol());
+                addProxiesPreparedStatement.setInt(3, serviceProxy.getServicePort());
+                addProxiesPreparedStatement.setInt(4, serviceProxy.getServiceBackendPort());
+                addProxiesPreparedStatement.setString(5, container.getBaseImageName());
+                addProxiesPreparedStatement.setString(6, container.getBaseImageVersion());
+                addProxiesPreparedStatement.setString(7, deploymentName);
+                addProxiesPreparedStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            String message = "Error while adding the database for service proxies to deployement : " + deploymentName;
+            log.error(message, e);
+        } finally {
+            AppFactoryDBUtil.closePreparedStatement(addProxiesPreparedStatement);
+        }
+
+    }
+
+    /**
+     * Get container
+     *
+     * @param deploymentId id of the deployment
+     * @param databaseConnection dabase connection
+     * @return list of containers
+     */
+    private List<Container> getContainers(int deploymentId, Connection databaseConnection) {
+        PreparedStatement getContainerPreparedStatement = null;
+        ResultSet containerResultSet = null;
+        List<Container> containers = new ArrayList<Container>();
+
+        try {
+            getContainerPreparedStatement = databaseConnection.prepareStatement(SQLConstants.GET_CONTAINER);
+            getContainerPreparedStatement.setInt(1, deploymentId);
+            containerResultSet = getContainerPreparedStatement.executeQuery();
+
+            //Adding container details
+            while (containerResultSet.next()) {
+                Container container = new Container();
+                container.setBaseImageName(
+                        containerResultSet.getString(SQLParameterConstants.COLUMN_NAME_BASEIMAGE_NAME));
+                container.setBaseImageVersion(
+                        containerResultSet.getString(SQLParameterConstants.COLUMN_NAME_BASEIMAGE_VERSION));
+
+                int containerId = containerResultSet.getInt(SQLParameterConstants.COLUMN_NAME_CONTAINER_ID);
+                List<ServiceProxy> serviceProxies = getServiceProxies(containerId, databaseConnection);
+                container.setServiceProxies(serviceProxies);
+                containers.add(container);
+            }
+
+        } catch (SQLException e) {
+            String message = "Error while getting containers";
+            log.error(message, e);
+        } finally {
+            AppFactoryDBUtil.closeResultSet(containerResultSet);
+            AppFactoryDBUtil.closePreparedStatement(getContainerPreparedStatement);
```

Comment:
I am closing the db connection after db commit.
https://github.com/wso2/product-af/pull/796/files#diff-b816ace5b656f21ef04cda5e862b059aR466